relayer/workers/parachaincommitmentrelayer/catchup.go

```diff
@@ -155,11 +157,31 @@ func (li *BeefyListener) parablocksWithProofs(blocks []ParaBlockWithDigest, late
 				return nil, err
 			}
 			li.log.WithField("relayBlockHash", relayBlockHash.Hex()).Info("Got relay chain blockhash")
-			allParaHeads, ownParaHeadProofPos, ownParaHead, err = li.relaychainConn.GetAllParaheadsWithOwn(relayBlockHash, li.paraID)
+			heads, err := li.relaychainConn.FetchParaHeads(relayBlockHash)
 			if err != nil {
 				li.log.WithError(err).Error("Failed to get paraheads")
 				return nil, err
 			}
+
+			if _, ok := heads[li.paraID]; !ok {
+				return nil, fmt.Errorf("chain is not a registered parachain")
```

Comment:
should this panic instead?

Reply:
Code in the relayer should never panic, as it will cause all workers to die. The idea with error-handling here is that the error will bubble up, the worker will self-terminate and be restarted.

Reply:
yeh - i guess in this case we know for sure the worker should not be restarted as there has most likely been a config issue, so it would be nice to be able to distinguish between error-that-should-restart and error-that-should-not but seems fine for now